editor/utils/with-history/index.js

```diff
@@ -74,6 +71,8 @@ const withHistory = ( options = {} ) => ( reducer ) => {
 					past: dropRight( past ),
 					present: last( past ),
 					future: [ present, ...future ],
+					lastAction: null,
```

Comment:
Just noting, that we are changing the existing logic for all action types. In the past, we would assign `action` instead. Isn't it what caused the issue rather than the fact that we were using instance variables?

I prefer to have them stored as they are in the updated version but I'm trying to understand what was the original issue.

Reply:
Yes, I know but this is a small change that doesn't have any impact but I think it's better this way because when you click the "undo/redo" you don't want to perform a comparison with these actions later, instead, you want to just record a new "past" when you start editing again